java/code/src/com/suse/manager/webui/controllers/virtualization/VirtualGuestsController.java

```diff
@@ -423,6 +427,23 @@ public ModelAndView console(Request request, Response response, User user) {
         return renderPage(request, response, user, "console", () -> data);
     }
 
+    /**
+     * Refresh the database with the actual list of virtual machines from the host
+     *
+     * @param request the request
+     * @param response the response
+     * @param user the user
+     * @return JSON result of the API call
+     */
+    public String refresh(Request request, Response response, User user) {
+        Server host = getServer(request, user);
+        VirtualInstanceManager.updateHostVirtualInstance(host,
+                VirtualInstanceFactory.getInstance().getFullyVirtType());
+        Optional<List<VmInfo>> plan = virtManager.getGuestsUpdatePlan(host.getMinionId());
+        plan.ifPresent(updatePlan -> VirtualInstanceManager.updateGuestsVirtualInstances(host, updatePlan));
+        return plan.isPresent() ? "Done" : "Failed";
```

Comment:
Why do you need to return a String instead of the Boolean value of the `isPresent()`?

Reply:
I wasn't aware we could return something else than a String in routes. Fixed